### ISSUE TITLE
[FEATURE REQUEST] getting region from EC2 instance metadata

### ISSUE THREAD

**Issue by pantocrator27** [fluent-plugins-nursery/fluent-plugin-cloudwatch-logs#73]
Would it be possible to get the region from EC2 instance metadata if it exists?

**Comment by mwpeterson** [comment]
On our k8s cluster, I hacked together my own solution with an initContainer to create an env file with the AWS_REGION from the metadata, and then a wrapper in the fluentd container to source the env file before starting fluentd.

**Comment by cosmo0920** [comment]
We don't like monolithic plugin.
How about using [fluent-plugin-ec2-matadata](https://github.com/takus/fluent-plugin-ec2-metadata) in your Fluentd pipeline?

**Comment by cosmo0920** [comment]
We decided that this issue won't fix.
Fluentd way is: Fluentd plugins should have single or not monolithic role.
EC2 instance metadata can be retrieve with https://github.com/takus/fluent-plugin-ec2-metadata and this plugin bundles filter plugin which is easily inserting data pipeline.

**Comment by skeggse** [comment]
How would we actually implement this? Trying to configure `region ${region}` results in "Invalid `:region` option was provided."
